chain/client/src/view_client.rs

```diff
@@ -438,7 +438,15 @@ impl ViewClientActor {
             .chain
             .get_block_header(&execution_outcome.transaction_outcome.block_hash)?])
         {
-            return Ok(TxExecutionStatus::Included);
+            return if execution_outcome
+                .receipts_outcome
+                .iter()
+                .all(|e| e.outcome.status != ExecutionStatusView::Unknown)
+            {
+                Ok(TxExecutionStatus::ExecutedOptimistic)
+            } else {
+                Ok(TxExecutionStatus::Included)
+            };
```

Comment:
This is the main change in this PR.
If the block for the tx is not finalised yet, let's see, maybe the receipts are ready. In this case, we can return ExecutedOptimistic.

If the block for tx is finalised, we have better options to return, we have Executed status.